src/Docs/Session/Retrieve/TaskResponse.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yoti\Docs\Session\Retrieve;
+
+abstract class TaskResponse
+{
+
+    /**
+     * @var string|null
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     */
+    private $state;
+
+    /**
+     * @var string|null
+     */
+    private $created;
+
+    /**
+     * @var string|null
+     */
+    private $lastUpdated;
+
+    /**
+     * @var GeneratedCheckResponse[]
+     */
+    private $generatedChecks = [];
+
+    /**
+     * @var GeneratedMedia[]
+     */
+    private $generatedMedia = [];
+
+    /**
+     * TaskResponse constructor.
+     * @param array<string, mixed> $task
+     */
+    public function __construct(array $task)
+    {
+        $this->id = $task['id'] ?? null;
+        $this->state = $task['state'] ?? null;
+        $this->created = $task['created'] ?? null;
+        $this->lastUpdated = $task['last_updated'] ?? null;
+
+        if (isset($task['generated_checks'])) {
+            $this->generatedChecks = $this->parseGeneratedChecks($task['generated_checks']);
+        }
+
+        if (isset($task['generated_media'])) {
+            $this->generatedMedia = $this->parseGeneratedMedia($task['generated_media']);
+        }
+    }
+
+    /**
+     * @param array<array<string, mixed>> $generatedChecks
+     * @return GeneratedCheckResponse[]
+     */
+    private function parseGeneratedChecks(array $generatedChecks): array
+    {
+        $parsedGeneratedChecks = [];
+        foreach ($generatedChecks as $generatedCheck) {
+            $parsedGeneratedChecks[] = new GeneratedCheckResponse($generatedCheck);
+        }
+        return $parsedGeneratedChecks;
+    }
+
+    /**
+     * @param array<array<string, mixed>> $generatedMediaArray
+     * @return GeneratedMedia[]
+     */
+    private function parseGeneratedMedia(array $generatedMediaArray): array
+    {
+        $parsedGeneratedMedia = [];
+        foreach ($generatedMediaArray as $generatedMedia) {
+            $parsedGeneratedMedia[] = new GeneratedMedia($generatedMedia);
+        }
+        return $parsedGeneratedMedia;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getId(): ?string
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getState(): ?string
+    {
+        return $this->state;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getCreated(): ?string
```

Comment:
Do we want to call this created_date instead? To make it a little more self-documenting?

Reply:
I think this will likely become a `DateTime` in PHP - we can just use `Yoti\Util\DateTime::stringToDateTime()`

Reply:
> Do we want to call this created_date instead? To make it a little more self-documenting?

I think Alex replied about this in the other PR, but we're generally just reflecting the JSON in these function names, so I think we were reluctant to deviate from them. 

As David mentions, it will give more clarity anyway when the type is a native DateTime object.